### DISCUSSION TITLE
Huge difference on request response time

### DISCUSSION THREAD

**Discussion by pablo-padua** [developmentseed/titiler#973] [Q&A] [answered]
Hi, I'm running a quick setup build following titiler's readme docs and I opted for the docker container run.
`
docker run --name titiler \
    -p 8000:8000 \
    --env PORT=8000 \
    --env WORKERS_PER_CORE=2 \
    --rm -it ghcr.io/developmentseed/titiler:latest
`
However, tile requests are taking forever and when compared to TiTiler's dev environment @ https://titiler.xyz it is insanely faster (6000ms X 100ms in general).

My question is, are there any after-container-up commands that must be run in order to make it faster, should I run the application directly and not via docker or what could possibly be the issue here ?

Via my VPS
![image](https://github.com/user-attachments/assets/0fa9ef9a-ad5e-4f1c-af5d-c60f25c60493)
Via TiTiler dev env
![image](https://github.com/user-attachments/assets/062e6edd-e9d5-465f-a119-42c6ecdef209)

**Comment by maxrjones** [comment]
One of the titiler devs may have a more detailed response for you, but FYI the GDAL environment variables referenced in [performance tuning](https://developmentseed.org/titiler/advanced/performance_tuning/) are known to [hugely influence tiling time](https://developmentseed.org/tile-benchmarking/02-run-tests/01-cog-gdal-tests.html).

**Reply by vincentsarago** [reply, marked as answer]
I think this might mostly be linked to the network. titiler.xyz is deployed on AWS Lambda using this configuration https://github.com/developmentseed/titiler/blob/725f857c735790e6c7783c4319f61804669b1980/deployment/aws/cdk/config.py#L20-L30

we use the same env in the docker compose file https://github.com/developmentseed/titiler/blob/725f857c735790e6c7783c4319f61804669b1980/docker-compose.yml#L24-L32

you may want to set those in your docker run command

**Reply by pablo-padua** [reply]
@vincentsarago and @maxrjones Awesome, thank you both for the reply.
Since you did say these were on the docker compose file, I ran
`git clone https://github.com/developmentseed/titiler.git`
`cd titiler`
`docker compose up --build titiler`
and I believe the previous way I was running it - was not using any of these commands because the get tile requests are much faster now.
Do you have some docs or guidelines that I should follow when exporting/transforming my .tiff to COG before uploading to BlobStorage/S3 for maximum performance?
I use GDAL.
Also, is there any way to improve the preview performance ? It's a few kb image it should query within a few ms but its taking 1.6s now (compared to the previous 6s that's gold but still could be better I guess)

**Reply by vincentsarago** [reply]
> Do you have some docs or guidelines that I should follow when exporting/transforming my .tiff to COG before uploading to BlobStorage/S3 for maximum performance?

depends what you want to do with your COG, if it's just for viz on a WebMercator map: https://cogeotiff.github.io/rio-cogeo/Advanced/#web-optimized-cog

> Also, is there any way to improve the preview performance ? It's a few kb image it should query within a few ms but its taking 1.6s now (compared to the previous 6s that's gold but still could be better I guess)

I have no idea 😬
